plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -42,4 +48,41 @@ public static boolean isManagedSdkFeatureEnabled() {
     }
     return false;
   }
+
+  /**
+   * Installs a Cloud SDK, if the preferences are configured to auto-manage the SDK. Blocks callers
+   * 1) if the managed SDK is being installed concurrently by others; and 2) until the installation
+   * is complete.
+   *
+   * @param consoleStream stream to which the install output is written
+   */
+  public static void installManagedSdk(MessageConsoleStream consoleStream)
+      throws CoreException, InterruptedException {
+    if (isManagedSdkFeatureEnabled()) {
+      if (CloudSdkPreferences.isAutoManaging()) {
+        runInstallJob(consoleStream, new CloudSdkInstallJob(consoleStream));
+      }
+    }
+  }
+
+  @VisibleForTesting
+  static void runInstallJob(MessageConsoleStream consoleStream, CloudSdkInstallJob installJob)
+      throws CoreException, InterruptedException {
+    installJob.schedule();
+    installJob.join();
+
+    IStatus status = installJob.getResult();
```

Comment:
IIRC if the job isn't a system job then a non-OK result will result in an error dialog…

Reply:
Perhaps we run it as a system job for local run and deploy, because they have the console and can cancel/stop in their ways. For the periodic install (not decided when to do), maybe a non-system job.